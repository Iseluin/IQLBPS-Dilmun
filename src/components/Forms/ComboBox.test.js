import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react'
import { ComboBox } from '.'

const options = [
    'test',
    'test1',
    'test2',
];

test('ComboBox renders correctly', () => {
    const tree = renderer.create(<ComboBox options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('should have a value', () => {
    render(<ComboBox value="test" options={options} />);
    const value = screen.getByDisplayValue(/test/i)
    expect(value).toBeTruthy();
})

test('should have a disabled', () => {
    render(<ComboBox disabled options={options} />);
    const input = screen.getByRole('combobox')
    expect(input).toBeDisabled();
})

