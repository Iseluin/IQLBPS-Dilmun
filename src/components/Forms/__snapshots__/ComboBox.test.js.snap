// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboBox renders correctly 1`] = `
<div
  className="w-full"
>
  <div
    className=" mt-4 relative"
  >
    <div
      className="relative w-full text-left bg-background text-primary font-medium rounded-lg shadow-md cursor-default focus:outline-none focus-visible:ring-2 focus-visible:ring-opacity-75 focus-visible:ring-background focus-visible:ring-offset-teal-300 focus-visible:ring-offset-2 sm:text-sm overflow-hidden border border-secondary"
    >
      <input
        aria-expanded={false}
        aria-multiselectable={true}
        className="w-full border-none focus:ring-0 py-2 pl-3 pr-10 text-sm leading-5 text-primary bg-background rounded-lg"
        data-headlessui-state=""
        disabled={false}
        id="headlessui-combobox-input-:r0:"
        onChange={[Function]}
        onCompositionEnd={[Function]}
        onCompositionStart={[Function]}
        onKeyDown={[Function]}
        role="combobox"
        type="text"
      />
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="absolute inset-y-0 right-0 flex items-center pr-2"
        data-headlessui-state=""
        disabled={false}
        id="headlessui-combobox-button-:r1:"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={-1}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-5 h-5 text-secondary"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10 3a1 1 0 01.707.293l3 3a1 1 0 01-1.414 1.414L10 5.414 7.707 7.707a1 1 0 01-1.414-1.414l3-3A1 1 0 0110 3zm-3.707 9.293a1 1 0 011.414 0L10 14.586l2.293-2.293a1 1 0 011.414 1.414l-3 3a1 1 0 01-1.414 0l-3-3a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
